pkg/kv/cosmosdb/store.go

```diff
@@ -0,0 +1,434 @@
+package cosmosdb
+
+import (
+	"context"
+	"encoding/base32"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/treeverse/lakefs/pkg/ident"
+	"github.com/treeverse/lakefs/pkg/kv"
+	kvparams "github.com/treeverse/lakefs/pkg/kv/params"
+)
+
+type Driver struct{}
+
+type Store struct {
+	containerClient  *azcosmos.ContainerClient
+	consistencyLevel azcosmos.ConsistencyLevel
+}
+
+const (
+	DriverName = "cosmosdb"
+)
+
+//nolint:gochecknoinits
+func init() {
+	kv.Register(DriverName, &Driver{})
+}
+
+// Open - opens and returns a KV store over CosmosDB. This function creates the DB session
+// and sets up the KV table.
+func (d *Driver) Open(ctx context.Context, kvParams kvparams.Config) (kv.Store, error) {
+	params := kvParams.CosmosDB
+	if params == nil {
+		return nil, fmt.Errorf("missing %s settings: %w", DriverName, kv.ErrDriverConfiguration)
+	}
+	if params.Endpoint == "" {
+		return nil, fmt.Errorf("missing endpoint: %w", kv.ErrDriverConfiguration)
+	}
+	if params.Database == "" {
+		return nil, fmt.Errorf("missing database: %w", kv.ErrDriverConfiguration)
+	}
+	if params.Container == "" {
+		return nil, fmt.Errorf("missing container: %w", kv.ErrDriverConfiguration)
+	}
+
+	log.Printf("CosmosDB: connecting to %s", params.Endpoint)
```

Comment:
This is the standard go log package - pass logger or avoid using this one

Reply:
Done.